File: java/parser/src/main/java/gov/lbl/parser/parser/Element_listVisitor.java
package gov.lbl.parser.parser;

import java.util.List;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Element;
import gov.lbl.parser.domain.Element_list;

import static java.util.stream.Collectors.toList;

public class Element_listVisitor extends modelicaBaseVisitor<Element_list> {
    @Override
    public Element_list visitElement_list(modelicaParser.Element_listContext ctx) {
        ElementVisitor elementVisitor = new ElementVisitor();
        List<Element> elements = ctx.element() == null ? null : ctx.element()
                .stream()
                .map(element -> element.accept(elementVisitor))
                .collect(toList());
        return new Element_list(elements);
    }
}

File: jsParser/parser/Element_listVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Element_list = require('../domain/Element_list')

const ElementVisitor = require('./ElementVisitor')

class Element_listVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitElement_list (ctx) {
    const elements = []

    if (ctx.element()) {
      const elementVisitor = new ElementVisitor.ElementVisitor()
      ctx.element().forEach(element => {
        elements.push(elementVisitor.visitElement(element))
      })
    }

    return new Element_list.Element_list(elements)
  }
};

Element_listVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitElement_list = this.visitElement_list
exports.Element_listVisitor = Element_listVisitor
